synapse/config/saml2_config.py

```diff
@@ -103,20 +141,14 @@ def read_config(self, config, **kwargs):
             saml2_config.get("saml_session_lifetime", "5m")
         )
 
-        mapping = saml2_config.get("mxid_mapping", "hexencode")
-        try:
-            self.saml2_mxid_mapper = MXID_MAPPER_MAP[mapping]
-        except KeyError:
-            raise ConfigError("%s is not a known mxid_mapping" % (mapping,))
-
     def _default_saml_config_dict(self):
         import saml2
 
         public_baseurl = self.public_baseurl
         if public_baseurl is None:
             raise ConfigError("saml2_config requires a public_baseurl to be set")
 
-        required_attributes = {"uid", self.saml2_mxid_source_attribute}
+        required_attributes = {"uid"}
```

Comment:
Note: I did this to uncouple mxid_source_attribute (which may not exist if the user has defined a custom module) and this block of code (which runs whether we're using a custom module or the default one).

I'm not sure how much it matters for that to be here.

Reply:
Have done so, seems like a sensible thing to do.